pillarbox-cast/src/main/java/ch/srgssr/pillarbox/cast/CastForwardingListener.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.cast
+
+import androidx.media3.common.AudioAttributes
+import androidx.media3.common.DeviceInfo
+import androidx.media3.common.FlagSet
+import androidx.media3.common.MediaItem
+import androidx.media3.common.MediaMetadata
+import androidx.media3.common.Metadata
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.PlaybackParameters
+import androidx.media3.common.Player
+import androidx.media3.common.Timeline
+import androidx.media3.common.TrackSelectionParameters
+import androidx.media3.common.Tracks
+import androidx.media3.common.VideoSize
+import androidx.media3.common.text.CueGroup
+
+/**
+ * Cast forwarding listener
+ *
+ * Because kotlin delegation with interface with default method are not handled correctly we have to implement all listener method.
+ *
+ * https://github.com/androidx/media/issues/2120
+ *
+ * @param player The player to forward to [onEvents].
+ * @param listener The [Player.Listener] to forward to.
+ */
+internal class CastForwardingListener(
+    private val player: Player,
+    private val listener: Player.Listener
+) : Player.Listener by listener {
+
+    override fun onTracksChanged(tracks: Tracks) {
+        // Do not forward this event.
+    }
+
+    override fun onAvailableCommandsChanged(availableCommands: Player.Commands) {
+        // Do not forward this event.
+    }
+
+    override fun onEvents(player: Player, events: Player.Events) {
+        // Filter Events triggered by CastPlayer
+        if (events.containsAny(Player.EVENT_AVAILABLE_COMMANDS_CHANGED, Player.EVENT_TRACKS_CHANGED)) {
+            return
+        }
+        val flagSet = FlagSet.Builder()
+            .apply {
+                for (index in 0 until events.size()) {
+                    val event = events.get(index)
+                    addIf(event, event != Player.EVENT_TRACKS_CHANGED && event != Player.EVENT_AVAILABLE_COMMANDS_CHANGED)
```

Comment:
Isn't `add(events.get(index))` enough here, since you check the kind of event on line 47?

Reply:
Good point, I think the return in line 47 is bad, if in the events there is a filtered event and some other event, we won't forward some vital event. I propose to remove the early return.